### ISSUE TITLE
Agent seems to be spawning multiple processes?

### ISSUE THREAD

**Issue by wpillar** [buildkite/buildbox-agent-ruby#18]
I'm using the following command to run the agent as a daemon (the wiki way didn't work):

```
start-stop-daemon -S -b -m -o -p /tmp/buildbox.piddeploy --exec /bin/bash -- -c  "/usr/local/bin/buildbox agent:start > /tmp/buildbox.logdeploy 2>&1"
```

What I've noticed is that the agent seems to be spawning multiple processes? Is this expected? It's causing issues with our SSH keys and private GitHub repos. 

Is this anything you're aware of or do you have any ideas?

**Comment by wpillar** [comment]
Seems to happen when triggering multiple rebuilds at once form the website. Shouldn't it queue them?

**Comment by keithpitt** [comment]
@wpillar yeah it should! Are you running multiple agents on the same machine perhaps?

**Comment by wpillar** [comment]
No, only got the one agent.

**Comment by keithpitt** [comment]
:( Yeah this is one of the problems with the ruby agent. It's super bloated, hard to maintain, and it has problems like this.

If you don't use artifacts, you can give the new agent a go: https://github.com/buildboxhq/agent-go

**Comment by keithpitt** [comment]
Official first version should be released today

**Comment by keithpitt** [comment]
@wpillar let me know if you try and, and I have to flip a feature for your account. Email keith@buildbox.io

**Comment by wpillar** [comment]
So the Go agent is the recommended one?
